witness/golang/cmd/witness/internal/witness/witness.go

```diff
@@ -206,6 +206,10 @@ func (w *Witness) Update(ctx context.Context, logID string, nextRaw []byte, proo
 		}
 		// If the proof is good store nextRaw and the new range.
 		r := []byte(log.Proof(nextRange).Marshal())
+		signed, err := w.signChkpt(nextNote)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't sign input checkpoint: %v", err)
+		}
 		if err := w.setChkptData(tx, logID, signed, r); err != nil {
 			return nil, fmt.Errorf("failed to store new checkpoint: %v", err)
 		}
```

Comment:
Could potentially factor this out into a separate func called from the 3 locations?